### PR TITLE
feat(offset): remove offset as a prop for Grid

### PR DESCRIPTION
**Pull request by arahansen**
offset is a nice-to-have but ultimately not necessary to implement all layouts we have come across
thus far. Removing offset will open the door for allowing the margin prop to use css margins

remove offset prop functionality

NOTE: it does seem to make some layouts slightly more awkward to produce. Usually requiring some sort of nested `Grid`/`Col` combo.

### CODE REVIEW

File: src/types.js [hunk 1]
```diff
@@ -2,8 +2,6 @@
 
 type zeroThroughEleven = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11
```
Comment (obartra) [inline src/types.js]: this is just used by `Size` now, we may as well move it there

File: storybook/stories/api/offset.js [hunk 2]
```diff
@@ -1,39 +0,0 @@
-// @flow
```
Comment (obartra) [inline storybook/stories/api/offset.js]: Do you think there is value in keeping this example but updating to use `<Offset />`?

File: storybook/stories/components/card/filter.js [hunk 3]
```diff
@@ -69,7 +74,9 @@ export default function() {
                 <ListItem index={4} />
 
                 <Grid>
-                  <p>{loremIpsum.substr(0, 100)}</p>
+                  <p>
+                    {loremIpsum.substr(0, 100)}
+                  </p>
```
Comment (obartra) [inline storybook/stories/components/card/filter.js]: have we verified if these chages are due to prettier version differences?

File: storybook/stories/components/header/app.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ export default function() {
   return (
     <Layout fixed="top" className={styles.colors1}>
       <Root align="center">
-        <Col align="center" size="auto" marginBottom={0}>
```
Comment (obartra) [inline storybook/stories/components/header/app.js]: why is `"auto"` wrong here?

File: storybook/stories/components/header/app.js [hunk 5]
```diff
@@ -49,8 +49,15 @@ export default function() {
...
         </Col>
-        <Col marginBottom={0} offset={1} size={3} className={styles.colors4}>
-          <input placeholder="Search for anything..." />
+        <Col
+          marginBottom={0}
+          justify="right"
+          size={3}
+          className={styles.colors4}
+        >
+          <Grid>
```
Comment (obartra) [inline storybook/stories/components/header/app.js]: No need for `Grid` here, is there?

File: storybook/stories/components/card/filter.js (hunk 3 above)
Comment (arahansen) [inline storybook/stories/components/card/filter.js]: Yeah I'm pretty sure my home laptop has a different version of the prettier plugin. Although I thought we had a pre commit hook that ran prettier? Which I would expect to be the same version for the project?

Comment (obartra) [inline storybook/stories/components/card/filter.js]: hm yeah that's weird, let's not worry about it for now we can tackle it later

File: storybook/stories/api/offset.js (hunk 2 above)
Comment (obartra) [inline storybook/stories/api/offset.js]: Yeah I think there's value to have it, particularly given that most 12 column grid systems offer it

**Pull request merged**